app/lib/dispose.ts

```diff
@@ -0,0 +1,51 @@
+import { EventPublisher, EventListener } from "./events";
+
+export default class Disposer {
```

Comment:
thoughts on using something like an rxjs subscription (perviously called "Disposable") https://rxjs.dev/guide/subscription

same below for all of that eventpublisher / eventnotifier stuff - feels like rxjs observables, you can do something like rxjs.fromEvent to create observables from dom events like click, etc

Reply:
I don't like the `addEventListener` / `removeEventListener` API and agree that the observable API is cleaner, but it's probably more performant if we stick to the JS EventTarget API (rxjs supports things like Node EventEmitter and JQuery objects, and also adds an extra wrapper layer in order to support the sub/unsub API).

In general when working on high-performance components, I tend to steer clear of external libs since there is too much going on under the hood that is hard to control.

Now that I think about it, I can actually delete `EventNotifier` class in favor of using `new EventTarget()` directly. (Done)